File: easy-vertx-web/src/main/java/io/github/shinglem/easyvertx/web/core/impl/RouteBase.java
package io.github.shinglem.easyvertx.web.core.impl;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author Shingle
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface RouteBase {

    /**
     * The value is used as a prefix for any route method declared on the class where {@link Route#path()} is used.
     *
     * @return the path prefix
     */
    String path() default "/";

    /**
     * The values are used for any route method declared on the class where {@link Route#produces()} is empty.
     *
     * @return the produced content types
     */
    String[] produces() default {};

    /**
     * The values are used for any route method declared on the class where {@link Route#consumes()} is empty.
     *
     * @return the consumed content types
     */
    String[] consumes() default {};

}

File: easy-vertx-web/src/test/kotlin/webtestexample/routebase/RouteBase.kt
package webtestexample.routebase

import io.github.shinglem.easyvertx.web.core.impl.Route
import io.github.shinglem.easyvertx.web.core.impl.RouteBase
import io.vertx.core.Vertx
import io.vertx.ext.web.Router

@RouteBase
class EmptyRouteBase {

    @Route
    fun emptyRoute() {

    }

    @Route(path = "/emptyRouteBase")
    fun emptyRouteBase(): String {
        return "emptyRouteBase"
    }

}

@RouteBase(path = "/notEmpty")
class NotEmptyRouteBase {

    @Route
    fun notEmpty(): String {
        return "notEmpty"
    }

    @Route(path = "/notEmptyTest")
    fun notEmptyTest(): String {
        return "notEmptyTest"
    }

}

@RouteBase(path = "/routes")
class Routes {

    @Route(path = "/1")
    @Route(path = "/2")
    fun notEmptyTest(): String = "routes"

}

